src/asset_util/src/lib.rs

```diff
@@ -465,21 +464,29 @@ fn collect_assets_rec(dir: &Dir, assets: &mut Vec<Asset>) {
     }
 }
 
-/// Returns the portion of the filename after the first dot.
-/// This corresponds to the file extension for the assets handled by this canister.
-///
-/// The builtin `extension` method on `Path` does not work for file extensions with multiple dots
-/// such as `.js.gz`.
-fn file_extension<'a>(asset: &'a File) -> &'a str {
-    asset
-        .path()
-        .file_name()
-        .unwrap()
-        .to_str()
-        .unwrap()
-        .split_once('.')
-        .unwrap()
-        .1
+/// Returns the extension of the given file, and the encoding type.
+/// If the text after the last dot is "gz" (i.e. this is a gzipped file), then the extension
+/// is the text after the second to last dot and the last dot in the file name
+/// (e.g. "js" for "some.gzipped.file.js.gz"), and the encoding is `ContentEncoding::GZip`.
+/// Otherwise the extension is the text after the last dot in the file name,
+/// and the encoding is `ContentEncoding::Identity`.
+/// This corresponds to the file extension for the assets handled by this crate.
+fn file_extension<'a>(asset: &'a File) -> (&'a str, ContentEncoding) {
```

Comment:
this is mixing concerns (grabbing the extension + figuring out the encoding); how do you feel about splitting the filename from extension, and if the extension is `.gz`, set the encoding to Gzip and check the extension on the rest of the filename?

Reply:
or alternatively return a truple of filename, content type & content encoding

Reply:
Thanks for the early feedback, indeed I've changed the helper to do the mapping to enums internally.